### ISSUE TITLE
manual_swap lint suggestion doesnt compile

### ISSUE THREAD

**Issue by antonilol** [rust-lang/rust-clippy#12334]
### Summary

title,
the code (see below) swaps two values, but the index of the second value is read from the array

a correct suggestion would be `let index = v[0]; v.swap(0, index);`

### Reproducer

I tried this code:

```rust
let mut v = [3, 2, 1, 0];

let index = v[0];
v[0] = v[index];
v[index] = index;
```

I expected to see this happen:
no clippy warning

Instead, this happened:
```
warning: this looks like you are swapping elements of `v` manually
  --> src/main.rs:66:5
   |
66 | /     let index = v[0];
67 | |     v[0] = v[index];
68 | |     v[index] = index;
   | |_____________________^ help: try: `v.swap(0, index);`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_swap
   = note: `#[warn(clippy::manual_swap)]` on by default
```

applying the suggestion `v.swap(0, index);` (using rust-analyzer in vscode) will cause it to fail compilation, applying it with clippy --fix will print that whole error text ("This likely indicates a bug in either rustc or cargo itself etc")

### Version

```text
rustc 1.76.0 (07dca489a 2024-02-04)
binary: rustc
commit-hash: 07dca489ac2d933c78d3c5158e3f43beefeb02ce
commit-date: 2024-02-04
host: x86_64-unknown-linux-gnu
release: 1.76.0
LLVM version: 17.0.6
```
